src/MasterSite.php

```diff
@@ -279,6 +279,37 @@ public function set_featured_image( $post_id, $post, $update ) {
 		}
 	}
 
+	/**
+	 * Generates hreflang metadata from countries.json template.
+	 */
+	public function generate_hreflang_meta() {
+
+		$request   = wp_remote_get( $this->theme_dir . '/templates/countries.json' );
+		$body      = wp_remote_retrieve_body( $request );
+		$countries = json_decode( $body, true );
+
+		$metadata = '';
+
+		foreach ( $countries as $key => $letter ) {
+			if ( 0 === $key ) {
+				$metadata .= "<link rel='alternate' hreflang='x-default' href='" . $letter[ $key ]['url'] . "'>\n";
+				continue;
+			}
+			foreach ( $letter as $country ) {
+				$lang = $country['lang'];
+				foreach ( $lang as $item ) {
+					if ( isset( $item['locale'] ) ) {
+						foreach ( $item['locale'] as $code ) {
+							$metadata .= "<link rel='alternate' hreflang='" . $code . "' href='" . $item['url'] . "'>\n";
```

Comment:
Since we're constructing html here, maybe it's easier to manage if we create a small twig template that loops over an array of data given to it? Then this function would be simpler as it would only need to filter the ones with a locale.

Reply:
Moved the html part in the template and just left the array construction here.